### DISCUSSION TITLE
urequests

### DISCUSSION THREAD

**Discussion by anahim09** [micropython/micropython#13765] [Using MicroPython]
When compiling my program I have this error, I don't know how to solve it. I already tried to import the library manually and in other ways, I would like it if you could help me

The program is as follows.

import utime  # Importa el módulo para controlar el tiempo
from machine import UART, Pin, ADC  # Importa los módulos para UART, Pin y ADC (conversor analógico a digital)
import urequests # Importa el módulo para realizar solicitudes HTTP

# Configuración de los pines y el UART
led = Pin(3, Pin.OUT)  # Configura el pin 3 como salida para controlar un LED
uart = UART(0, baudrate=115200, tx=Pin(0), rx=Pin(1))  # Configura el UART con baudrate de 115200, y los pines TX y RX

# Configuración del sensor de temperatura
sensor_temp = ADC(4)  # Configura el pin 4 como entrada analógica para el sensor de temperatura
factor = 3.3 / 65535  # Factor de conversión para convertir lecturas analógicas a voltaje

# Función para enviar comandos AT al módulo UART
def send_at_command(command):
    uart.write(command + '\r\n')  # Envía el comando AT seguido de retorno de carro y nueva línea
    utime.sleep(1)  # Espera un momento para la respuesta del módulo
    response = uart.read()  # Lee la respuesta del módulo UART
    if response:
        print("Respuesta del módulo UART:", response)  # Imprime la respuesta del módulo UART si existe

# Bucle principal del programa
while True:
    lectura = sensor_temp.read_u16() * factor  # Lee el valor del sensor de temperatura y lo convierte a voltaje
    temperatura = 27 - (lectura - 0.706) / 0.001721  # Calcula la temperatura en base al voltaje del sensor

    # Envía la temperatura medida a un servidor web utilizando la API ThingSpeak
    enviar = urequests.get("https://api.thingspeak.com/update?api_key=FS5G8WSBZ4XK3GYX&field1=0")

    # Verifica si la temperatura está fuera del rango deseado y envía un mensaje al módulo UART si es necesario
    if temperatura < 20 or temperatura > 25:
        send_at_command("AT+SEND=1,36,Temperatura fuera de rango; MODULO 1")
        utime.sleep(5)  # Espera 5 segundos antes de continuar
        
    # Lee datos del UART y verifica si hay datos y si no están en el formato esperado
    data = uart.read()
    if data and len(data) != 20:
        print(data + f" La temperatura del sensor es: {temperatura} TEMPERATURA FUERA DE RANGO")
    else:
        print(f"La temperatura del sensor es: {temperatura}")
        data = None  # Restablece la variable de datos a None

    utime.sleep(5)  # Espera 5 segundos antes de la próxima iteración del bucle

And the error is this

Traceback (most recent call last):
  File "<stdin>", line 28, in <module>
  File "urequests.py", line 180, in get
  File "urequests.py", line 76, in request
OSError: -6

**Comment by karfas** [comment]
Please read [Welcome (please read before posting)](https://github.com/orgs/micropython/discussions/9111) regarding code formatting.

Most likely, thingspeak.com sends some error response, the request times out  or you are running out of memory.
You might find some hints when searching for "urequest" in this discussion forum.

**Reply by anahim09** [reply]
this error occurs with any url:(

**Reply by karfas** [reply]
> this error occurs with any url:(

Most likely some kind of network problem.
Firewall ?
Routing ?
DNS problem ? Try IP addresses.
Can you ping your device ?
Can you reach local (on **your** network) hosts ?
Are you sure the device is connected to the WLAN ?
